packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1079,7 +1079,8 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
     super.didUpdateWidget(oldWidget);
     if (_isListeningToStream &&
         (widget.loadingBuilder == null) != (oldWidget.loadingBuilder == null)) {
-      _imageStream.removeListener(_getListener(oldWidget.loadingBuilder));
+      _imageStream.removeListener(_getListener());
+      _resetListener();
```

Comment:
There are a few other places in this class where we remove the listener. It seems like in all of them we would want to reset the listener right? Why not just make reset listener remove the old listener if it's not null, and then null out the old listener, and then use that in all the spots where we currently call `removeListener`?

Reply:
This is the only place where you want to replace the listener with a new one, but I guess it doesn't hurt to combine the calls. I'll check the other calls to removeListener.

Reply:
Actually, merging resetListener is a confusing method overall. The problem is that you also have the boolean for _isListeningToStream, and you don't want to touch that one. 

I replaced _resetListener with an optional parameter on _getListener: `_getListener({bool recreateListener = false})`. Now it just does what it says, recreating the listener and nothing else.